### PR TITLE
fix: keep agent prompt cache prefix stable

### PR DESCRIPTION
**Pull request by danny-avila**
## Summary
- split AgentContext system content into a stable cacheable prefix and dynamic additional_instructions tail
- keep Anthropic cache_control only on stable system text and place Bedrock cache points before dynamic text
- treat cross-run summaries as dynamic system tail while preserving mid-run compaction as HumanMessage
- add opt-in live Anthropic and Bedrock prompt-cache tests that verify cache creation/read behavior across dynamic tails
- bump @librechat/agents to 3.1.75

## Tests
- npm test -- src/agents/__tests__/AgentContext.test.ts --runInBand
- RUN_ANTHROPIC_PROMPT_CACHE_LIVE_TESTS=1 RUN_BEDROCK_PROMPT_CACHE_LIVE_TESTS=1 npm test -- src/agents/__tests__/AgentContext.anthropic.live.test.ts src/agents/__tests__/AgentContext.bedrock.live.test.ts src/agents/__tests__/AgentContext.test.ts --runInBand
- npm run build
- npm pack --json

## Optional live verification
- RUN_ANTHROPIC_PROMPT_CACHE_LIVE_TESTS=1 ANTHROPIC_API_KEY=... npm test -- AgentContext.anthropic.live.test.ts --runInBand
- RUN_BEDROCK_PROMPT_CACHE_LIVE_TESTS=1 BEDROCK_AWS_REGION=... BEDROCK_AWS_ACCESS_KEY_ID=... BEDROCK_AWS_SECRET_ACCESS_KEY=... npm test -- AgentContext.bedrock.live.test.ts --runInBand

### CODE REVIEW

**Comment by danny-avila** [comment]
@codex review

**Comment by danny-avila** [comment]
@codex review

File: src/messages/cache.ts [hunk 1]
```diff
@@ -314,6 +314,14 @@ export function addBedrockCacheControl<
       'getType' in originalMessage &&
       typeof originalMessage.getType === 'function' &&
       originalMessage.getType() === 'tool';
+    const isSystemMessage =
+      'getType' in originalMessage &&
+      typeof originalMessage.getType === 'function' &&
+      originalMessage.getType() === 'system';
```
[the remark this answers is not in the document]
Comment (danny-avila) [reply src/messages/cache.ts]: Fixed in 9ed8de7. addBedrockCacheControl now skips system messages by either LangChain getType() or serialized role, and only computes the message kind once. Added regression coverage for both LangChain SystemMessage and plain { role: "system" } messages preserving existing system cachePoint/dynamic-tail content unchanged. Also covered dynamic-only prompt-cache edge cases and reran lint/typecheck plus unit and live Anthropic/Bedrock prompt-cache tests.

**Comment by danny-avila** [comment]
@codex review

**Comment by danny-avila** [comment]
@codex review

**Comment by danny-avila** [comment]
@codex review

File: src/messages/cache.ts [hunk 2]
```diff
@@ -310,11 +310,21 @@ export function addBedrockCacheControl<
...
+        ? originalMessage.getType()
+        : undefined;
+    const messageRole =
+      'role' in originalMessage && typeof originalMessage.role === 'string'
+        ? originalMessage.role
+        : undefined;
+
+    if (messageType === 'system' || messageRole === 'system') {
+      continue;
+    }
```
[the remark this answers is not in the document]
Comment (danny-avila) [reply src/messages/cache.ts]: Fixed in 6e6b8b0. addBedrockCacheControl still avoids adding or moving cache points in system messages, but it now strips Anthropic cache_control fields from array-based system content before Bedrock conversion. Added regressions for both LangChain SystemMessage and serialized { role: "system" } messages to preserve existing Bedrock cachePoint placement while removing incompatible Anthropic cache markers. Local checks: eslint on touched cache files, cache.test.ts, AgentContext.test.ts, and tsc --noEmit.

**Comment by danny-avila** [comment]
@codex review

**Comment by danny-avila** [comment]
@codex review
